src/core/main.c

```diff
@@ -2799,6 +2799,10 @@ int main(int argc, char *argv[]) {
                                         goto finish;
                                 }
 
+                                /* We might have just mounted /proc, so let's try to parse the kernel
+                                 * command line log arguments immediately. */
+                                log_parse_environment();
```

Comment:
this looks like a much better place.

however one more tweak please: please end the `if (!skip_setup)` block immediately before this call, and start a new block after the `log_open()` immediately below.

(background: the skip_setup boolean is true during the initial boot, but false on reexec. It's supposed to guard everything we only should do once when we first boot, but not what we are supposed to do when we reexec. parsing the log environment and subsequently opening the log sockets is something we should do on reexec, hence it should not be conditionalized like that.)